### PR TITLE
Prefer to not prapagate marker REMOTE_TASK_FAILED failure externally

### PR DESCRIPTION
**Pull request by losipiuk**

### CODE REVIEW

**Comment by losipiuk** [comment]
Approach to fix https://github.com/trinodb/trino/issues/10395.

File: core/trino-main/src/main/java/io/trino/operator/DeduplicationExchangeClientBuffer.java [hunk 1]
```diff
@@ -266,6 +267,12 @@ private synchronized void checkInputFinished()
                         if (failure == null) {
                             failure = taskFailure;
                         }
+                        else if (failure instanceof TrinoException && ((TrinoException) failure).getErrorCode().equals(REMOTE_TASK_FAILED.toErrorCode()) && failure != taskFailure)
+                        {
+                            // prefer to not prapagate marker REMOTE_TASK_FAILED failure externally
```
Comment (losipiuk) [inline core/trino-main/src/main/java/io/trino/operator/DeduplicationExchangeClientBuffer.java:272]: @arhimondr I think it is not bullet proof as we may not have learned yet about actually interesting task failure when we are building final exception. Right?

Comment (arhimondr) [reply core/trino-main/src/main/java/io/trino/operator/DeduplicationExchangeClientBuffer.java:272]: Yeah, I wonder if all of the `REMOTE_TASK_FAILED` should be ignored?

**Comment by losipiuk** [comment]
Naa - does not make sense - need more work.

**Pull request closed without merging**